### ISSUE TITLE
feat: adding support for go and lua parent nodes

### ISSUE THREAD

**Issue by arnevm123** [ThePrimeagen/refactoring.nvim#434]
Hi,
I noticed some languages have support for using parent nodes in the print statement. 
I've implemented this locally for go and lua and I'd like to make a PR.
Any pointers on how to make sure I got all cases?
I can make a PR with the ones I've found if interested.

lua
```
local special_nodes = {
    "method_index_expression",
    "dot_index_expression",
}
```

go
```
local special_nodes = {
	"selector_expression",
}
```

**Comment by TheLeoP** [comment]
Hi c:

Could you provide some example code for the current behavior and your expected behavior for these nodes?

**Comment by arnevm123** [comment]
Lua:
```
local foo = {
	bar = "hello",
}
foo.bar = "test"
--  CURRENT
-- __AUTO_GENERATED_PRINT_VAR_START__
print(" bar:", vim.inspect(bar)) -- __AUTO_GENERATED_PRINT_VAR_END__
-- FIXED
-- __AUTO_GENERATED_PRINT_VAR_START__
print(" foo.bar:", vim.inspect(foo.bar)) -- __AUTO_GENERATED_PRINT_VAR_END__
.
```

Go:
```
package main

import (
	"fmt"
)

type foo struct {
	bar string
}

func main() {
	test := foo{}
	test.bar = "hello"
	// ORIGINAL:
	// __AUTO_GENERATED_PRINT_VAR_START__
	fmt.Printf("main bar: %v \n", bar) // __AUTO_GENERATED_PRINT_VAR_END__

	// FIX:
	// __AUTO_GENERATED_PRINT_VAR_START__
	fmt.Printf("main test.bar: %v \n", test.bar) // __AUTO_GENERATED_PRINT_VAR_END__
}
```

**Comment by TheLeoP** [comment]
> Any pointers on how to make sure I got all cases?

I think that, just like with refactoring captures, this should be supported in an best effor basis. If a user finds a new case, they can open an Issue just like this one.

> I've implemented this locally for go and lua and I'd like to make a PR.

I missread this the first time I looked at this issue, sorry. I thought you were asking for guidance on how to make this PR and, since it was a simple change, I intended to create the PR myself.

> I noticed some languages have support for using parent nodes in the print statement.

This isn't used exclusively by printf_var, so I'd like to test if something breaks
